packages/core/src/lib/input/input.svelte

```diff
@@ -13,33 +13,92 @@ NumberInput or DateInput are preferable.
 -->
 <svelte:options immutable />
 
+<script
+  lang="ts"
+  context="module"
+>
+export type InputState = 'info' | 'warn' | 'error' | 'none';
+</script>
+
 <script lang="ts">
+import Icon from '$lib/icon/icon.svelte';
 import cx from 'classnames';
 
+export let value = '';
+
 /** Whether or not the input should be rendered as readonly and be operable. */
 export let readonly = false;
 
 /** Whether or not the input should be rendered as readonly and be non-operable. */
 export let disabled = false;
-</script>
 
-<input
-  {...$$restProps}
-  readonly={disabled || readonly ? true : undefined}
-  aria-disabled={disabled ? true : undefined}
-  class={cx(
-    'h-[30px] w-full appearance-none border px-2 py-1.5 text-xs leading-tight outline-none',
-    {
-      'border-disabled-light bg-disabled-light text-disabled-dark':
-        disabled || readonly,
-      'pointer-events-none': disabled,
+/** The state of the input (info, warn, error, success), if any. */
+export let state: InputState = 'none';
+
+$: isInfo = state === 'info';
+$: isWarn = state === 'warn';
+$: isError = state === 'error';
+
+let icon = '';
+$: {
+  switch (state) {
+    case 'info': {
+      icon = 'information';
+      break;
+    }
+    case 'warn': {
+      icon = 'alert';
+      break;
     }
-  )}
-  on:input
-  on:keydown
-  on:focus
-  on:blur
-/>
+    case 'error': {
+      icon = 'alert-circle';
+      break;
+    }
+    default: {
+      icon = '';
+      break;
+    }
+  }
+}
```

Comment:
this is a cool pattern that's a bit more compact:
```suggestion
$: icon = {
  info: 'information',
  warn: 'alert'
  error: 'alert-circle'
}[state] ?? ''
```
but either works for me!